client/web/src/enterprise/repo/settings/RepoSettingsPermissionsPage.tsx

```diff
@@ -81,10 +147,104 @@ export const RepoSettingsPermissionsPage: FC<RepoSettingsPermissionsPageProps> =
+                        aria-label="Search users..."
+                        variant="regular"
+                    />
+                    <div className="flex-1" />
+                </div>
+                <Table<INode> columns={TableColumns} getRowId={node => node.id} data={connection.nodes} />
+                <PageSwitcher {...paginationProps} totalCount={null} className="mt-4" />
+            </Container>
         </>
     )
 }
 
+const TableColumns: IColumn<INode>[] = [
+    {
+        key: 'user',
+        header: 'User',
+        render: ({ user }: INode) => user && <RenderUsernameAndEmail key={user.id} user={user} />,
+    },
+    {
+        key: 'reason',
+        header: { label: 'Reason', align: 'center' },
+        align: 'center',
+        render: ({ reason }: INode) => (
+            <div className="d-flex justify-content-center">
+                <Badge {...(PermissionReasonBadgeProps[reason] || {})}>{reason}</Badge>
+            </div>
+        ),
+    },
+    {
+        key: 'updatedAt',
+        header: { label: 'Updated At', align: 'center' },
+        align: 'center',
+        render: ({ updatedAt }: INode) => (
+            <div className={classNames('d-flex justify-content-center', styles.updatedAtCell)}>
+                {updatedAt ? <Timestamp date={updatedAt} /> : '-'}
+            </div>
+        ),
+    },
+]
+
+export const RenderUsernameAndEmail: FC<{ user: IUser }> = ({ user }) => {
+    const [isOpen, setIsOpen] = useState<boolean>(false)
+    const handleOpenChange = useCallback((event: PopoverOpenEvent): void => {
+        setIsOpen(event.isOpen)
+    }, [])
+
+    if (!user) {
+        return null
+    }
+
+    return (
+        <div className="d-flex p-2 align-items-center">
+            <UserAvatar inline={true} user={user} />
+            <Link to={`/users/${user.username}`} className="text-truncate ml-2">
+                @{user.username}
+            </Link>
+            <Popover isOpen={isOpen} onOpenChange={handleOpenChange}>
+                <PopoverTrigger as={Button} className="ml-1 border-0 p-1" variant="secondary" outline={true}>
+                    <Icon aria-label="Show details" svgPath={mdiChevronDown} />
+                </PopoverTrigger>
+                <PopoverContent position={Position.bottom} focusLocked={false}>
+                    <div className="p-2">
+                        <Text className="mb-0">{user.displayName}</Text>
+                        <Text className="mb-0">{user.email}</Text>
+                    </div>
+                </PopoverContent>
+            </Popover>
+        </div>
+    )
+}
+
+const PermissionReasonBadgeProps: { [reason: string]: BadgeProps } = {
+    'Permissions Sync': {
+        variant: 'success',
+        tooltip: 'The repository is accessible to the user due to permissions syncing from code host.',
+    },
+    Unrestricted: { variant: 'primary', tooltip: 'The repository is unrestricted and accessible to all the users. ' },
```

Comment:
nit: I would remove repetition
```suggestion
    Unrestricted: { variant: 'primary', tooltip: 'The repository is  accessible to all the users. ' },
```